kong/plugins/grpc-web/handler.lua

```diff
@@ -0,0 +1,91 @@
+-- Copyright (c) Kong Inc. 2020
+
+local deco = require "kong.plugins.grpc-web.deco"
+
+local ngx = ngx
+local kong = kong
+
+local string_format = string.format
+local ngx_req_get_method = ngx.req.get_method
+local kong_request_get_path = kong.request.get_path
+local kong_request_get_header = kong.request.get_header
+local kong_request_get_raw_body = kong.request.get_raw_body
+local kong_response_exit = kong.response.exit
+local kong_response_set_header = kong.response.set_header
+local kong_service_request_set_header = kong.service.request.set_header
+local kong_service_request_set_raw_body = kong.service.request.set_raw_body
+
+
+local grpc_web = {
+  PRIORITY = 1,
+  VERSION = '0.1.0',
+}
+
+
+function grpc_web:access(conf)
+  kong_response_set_header("Access-Control-Allow-Origin", "*")
+
+  if ngx_req_get_method() == "OPTIONS" then
+    return kong_response_exit(200, "OK", {
```

Comment:
This static table can be cached to reduce runtime GC pressure.

Reply:
done.  I'll probably (partially) undo at some point, since i think there a better answer (maybe with non-static fields) can help the browser to omit subsequent CORS queries.